test/io/test_octree_compression.cpp

```diff
@@ -48,102 +48,182 @@ int total_runs = 0;
+        pcl::PointCloud<pcl::PointXYZ>::Ptr cloud(new pcl::PointCloud<pcl::PointXYZ>());
         // assign input point clouds to octree
         // create random point cloud
         for (int point = 0; point < point_count; point++)
         {
-          // gereate a random point
-          pcl::PointXYZRGBA new_point;
-          new_point.x = static_cast<float> (MAX_XYZ * rand() / RAND_MAX);
-          new_point.y = static_cast<float> (MAX_XYZ * rand() / RAND_MAX),
-          new_point.z = static_cast<float> (MAX_XYZ * rand() / RAND_MAX);
-          new_point.r = static_cast<int> (MAX_COLOR * rand() / RAND_MAX);
-          new_point.g = static_cast<int> (MAX_COLOR * rand() / RAND_MAX);
-          new_point.b = static_cast<int> (MAX_COLOR * rand() / RAND_MAX);
-          new_point.a = static_cast<int> (MAX_COLOR * rand() / RAND_MAX);
-          // OctreePointCloudPointVector can store all points..
+          // generate a random point
+          pcl::PointXYZ new_point(static_cast<float> (MAX_XYZ * rand() / RAND_MAX),
+                                 static_cast<float> (MAX_XYZ * rand() / RAND_MAX),
+                                 static_cast<float> (MAX_XYZ * rand() / RAND_MAX));
           cloud->push_back(new_point);
         }
-
-//        std::cout << "Run: " << total_runs << " compression profile:" << compression_profile << " point_count: " << point_count;
+        EXPECT_EQ(cloud->height, 1);
+  //      std::cout << "Run: " << total_runs << " compression profile:" << compression_profile << " point_count: " << point_count;
         std::stringstream compressed_data;
-        pointcloud_encoder.encodePointCloud(cloud, compressed_data);
-        pointcloud_decoder.decodePointCloud(compressed_data, cloud_out);
-        EXPECT_GT((int)cloud_out->width, 0) << "decoded PointCloud width <= 0";
-        EXPECT_GT((int)cloud_out->height, 0) << " decoded PointCloud height <= 0 ";
-      }
-      catch (std::exception& e)
-      {
-        std::cout << e.what() << std::endl;
-      }
-    } // runs
-  } // compression profiles
+        try
+        { // decodePointCloud() throws exceptions on errors
+          pointcloud_encoder.encodePointCloud(cloud, compressed_data);
+          pointcloud_decoder.decodePointCloud(compressed_data, cloud_out);
+          if (pcl::io::compressionProfiles_[compression_profile].doVoxelGridDownSampling) {
+            EXPECT_GT(cloud_out->width, 0);
+            EXPECT_LE(cloud_out->width, cloud->width) << "cloud width after encoding and decoding greater than before. Profile: " << compression_profile;
+          }
+          else {
+            EXPECT_EQ(cloud_out->width, cloud->width) << "cloud width after encoding and decoding not the same. Profile: " << compression_profile;
+          }
+          EXPECT_EQ(cloud_out->height, 1) << "cloud height after encoding and decoding should be 1 (as before). Profile: " << compression_profile;
+        }
+        catch (std::exception& e)
```

Comment:
Can we catch some specific error here? What error are we expecting to be thrown?

Reply:
Not sure. One comment says that `decodePointCloud` might throw an exception, but I don't see where or when it would do that. Just printing the message to `cout` where it is likely not noticed is a bad idea anyways, and in my test runs, there was never an exception. So I would just remove the try-catch constructs?

Reply:
I removed the try-catch statements.